beautifultable/beautifultable.py

```diff
@@ -1415,3 +1416,87 @@ def get_string(self, recalculate_width=True):
             string_.append(line)
 
         return "\n".join(string_)
+
+    def to_csv(self, file_name='beautiful_table.csv', delimiter=','):
+        """Export table to CSV format.
+
+        Parameters
+        ----------
+        file_name : str, optional
+            Path to CSV file which BeautifulTable will write to.
+
+        delimiter : str, optional
+            Delimiter used as value separator. Defaults to comma (',').
+        """
+        with open(file_name, mode='wt', newline='') as csv_file:
+            csv_writer = csv.writer(csv_file,
+                                    delimiter=delimiter,
+                                    quoting=csv.QUOTE_MINIMAL)
+            csv_writer.writerow(self.column_headers)  # write header
+            csv_writer.writerows(self._table)  # write table
+
+    def _convert_row(self, row):
+        """Helper function for parsing elements of CSV rows.
+
+        Row that contains integer or float values are automatically
+        converted to string when read from CSV file.
+        This function returns row while keeping proper types on row
+        elements.
+
+        Parameters
+        ----------
+        row : list[str]
+            Row read from CSV file.
+
+        Returns
+        -------
+        list[str]
+            Row with proper types.
+        """
+        for element_index in range(len(row)):
+            try:
+                a = float(row[element_index])
+            except ValueError:
+                continue
+            else:
+                row[element_index] = int(a) if a.is_integer() else a
+
+        return row
+
+    def from_csv(self, file_name, delimiter=','):
+        """Create table from CSV file.
+
+        Parameters
+        ----------
+        file_name : str
+            Path to CSV file which `BeautifulTable` will read from.
+        delimiter : str, optional
+            Delimiter used as value separator. Defaults to comma (`,`).
+
+        Raises
+        ------
+        ValueError
+            If `file_name` is not str type.
+        FileNotFoundError
+            If `file_name` is not valid path to file.
+        """
+
+        if not isinstance(file_name, str):
+            raise ValueError(
+                (
+                    "Expected 'file_name' to be string, got {}"
+                ).format(type(file_name).__name__)
+            )
+
+        try:
+            with open(file_name, mode='rt', newline='') as csv_file:
+                csv_file = csv.reader(csv_file, delimiter=delimiter)
+                table = BeautifulTable()
```

Comment:
Instead of creating another BeautifulTable instance, I think it makes more sense to add that data to the already exisitng table. Allows user to apply styling only once if there are multiple files to be read.

Reply:
I didn't quite understood this one. I think i understood the first part, to apply data to current created `BeautifulTable` object, not to return new instance. But the second part, what styling and multiple files to you mean ?

Reply:
Suppose I already have a table object with various styling like alignment, padding, borders, etc created. I can then reuse that table for importing csv files with similar data. I can then clear the table and again reuse it if I wish to. However If you return a new table instance, user would have to apply all those changes to the new instance too.

Reply:
Ok. I understand now :+1: